LFS-179 As a user, I can record chemotherapy treatment details

prabnur opened the pull request.

prabnur committed:
LFS-179 Generate clinical data questionnaire for Chemotherapy

prabnur committed:
LFS-168 / LFS-179 Refactor JSON file

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+    "text": "Patient ID",
+    "minAnswers": 1,
+    "maxAnswers": 1,
+    "displayMode": "hidden"
```
I'm not sure that this nor the Tumor number should be hidden? Shouldn't these be enterable by the user?

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+  "Diagnosis (Morphology) as on pathology report": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Diagnosis (Morphology) as on pathology report",
+    "displayMode": "hidden",
```
Again -- why is this hidden?

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+    "minAnswers": 1,
+    "maxAnswers": 1,
+    "displayMode": "list",
+    "dataType": "text",
```
This sounds more like dataType="boolean" but we don't yet handle NA as a possible boolean -- is this something we should look into implementing @sdumitriu ?

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+  },
+  "Weight at start of protocol (kg)": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Weight at start of protocol (kg)",
```
This should use the new `unitOfMeasurement` option instead of putting it in the text, as per #119 .

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+  },
+  "Height at start of protocol (cm)": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Height at start of protocol (cm)",
```
`unitOfMeasurement` again.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+  },
+  "Start date (yyyy-mm-dd)": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Start date (yyyy-mm-dd)",
```
We don't need to specify the date format in the question text, since it's handled by the UI element.
Moreover, why is `minAnswers` 0?

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+  },
+  "End date (yyyy-mm-dd)": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "End date (yyyy-mm-dd)",
```
As before, we don't need to  specify the date format in the question text, since it's handled by the UI element.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+  },
+  "Total duration if continuous (months)": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Total duration if continuous (months)",
```
Switch to `unitOfMeasurement`.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+  },
+  "Cumulative dose (x/m2)": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Cumulative dose (x/m2)",
```
Switch to `unitOfMeasurement`. Does not giving an answer here mean that they've just started treatment?

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+    "minAnswers": 1,
+    "maxAnswers": 1,
+    "displayMode": "list",
+    "dataType": "text",
```
This should be a `boolean` question. Also remove the info in the brackets of the text.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,180 @@
+  },
+  "protocol information": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "protocol information",
```
Why's this not capitalized?

OrdiNeu reviewed (dismissed):
Looks good, though I'm not 100% clear on some of the details here. @sdumitriu should we add a fourth state for our `BooleanQuestion`s (Y/N/Unknown/NA)?

marta- committed:
LFS-179 As a user, I can record chemotherapy treatment details
- Removed questions regarding patient id, tumor number, and diagnosis, which will be pre-filled based on the linked subject id
- Renamed question field ids
- Changed question type from text to boolean where appropriate
- Removed unnecessary displayType where it matched the default
- Specified unitOfMeasurement where applicable
- Cleaned up question texts (capitalized, removed andswer format and unit of measurement from text).

marta- reviewed (approved):
Works fine after the latest fixes.

sdumitriu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,126 @@
+    "text": "Weight at start of protocol",
+    "minAnswers": 1,
+    "maxAnswers": 1,
+    "dataType": "decimal",
```
Should have a min value, since `-12` is not a very probable weight. Same for all the other number-type fields that record a real measurable value.

sdumitriu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,126 @@
+    "text": "Chemotherapy Agent",
+    "minAnswers": 1,
+    "maxAnswers": 1,
+    "dataType": "vocabulary",
```
I thought there are no more vocabularies in the first version...

marta- replied:
I removed vocabulary mappings from #113 (LFS-178 tumor data) because there was some oambiguity what mapppes where. This can stay as is.

sdumitriu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,126 @@
+    "dataType": "date",
+    "dateFormat": "yyyy-MM-dd"
+  },
+  "end_date": {
```
This could use a `validate: > start_date` setting, too bad we don't yet support that...

sdumitriu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,126 @@
+    "jcr:primaryType": "lfs:Question",
+    "text": "Mode",
+    "minAnswers": 1,
+    "maxAnswers": 1,
```
The validation message is misleading, "Please select _at least_ 1 option". @OrdiNeu since exactly one is expected, the "at least" part should be removed.

marta- replied:
Created https://phenotips.atlassian.net/browse/LFS-257 , to be addressed separately.

sdumitriu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,126 @@
+    "minAnswers": 0,
+    "maxAnswers": 1,
+    "dataType": "decimal",
+    "unitOfMeasurement" : "/m²"
```
That unit seems wrong... Is this really the correct unit, or is something missing?

sdumitriu replied:
The spreadsheet has `x/m2`, and the value must actually say if it is `mg` or `g`, and some of the value are really free text like `120d*50mg*3/d= 18,000 (not sure!)` or `98mg/1.62 (approximate BSA) = 60.49` (what does `BSA` mean?). Should we instead use `mg/m2` as the unit? Or should we turn this into a free text field?

marta- replied:
Changing to mg/m2.

sdumitriu replied:
`BSA` = `Body Surface Area`

sdumitriu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Chemotherapy.json:
```diff
@@ -0,0 +1,126 @@
+    "text": "Dose reduction",
+    "minAnswers": 1,
+    "maxAnswers": 1,
+    "dataType": "boolean",
```
Well, there are many other values entered in this column...

marta- replied:
They are misplaced, they should be stored in the `Comments` when we import the data.

marta- committed:
LFS-179: As a user, I can record chemotherapy treatment details
[Addressing review comments]
- Added min value to numeric fields
- corrected dose unit of measurement
